apiclient/httpapiclient.go

```diff
@@ -0,0 +1,52 @@
+package apiclient
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+var instance *HttpApiClient
+
+type HttpApiClient struct {
+	client *http.Client
+}
+
+func GetHttpApiClientInstance() *HttpApiClient {
+	once := sync.Once{}
+	once.Do(func() {
+		client := &http.Client{
+			Timeout: time.Second * 10,
+		}
+		instance = &HttpApiClient{client: client}
+	})
+
+	return instance
+}
+
+func (api *HttpApiClient) Get(url string, params map[string]interface{}, response interface{}) error {
+	resp, err := api.client.Get(url)
```

Comment:
You can actually mock this client and that would be useful for unit testing this.

Since you already know that you need a client struct that implements the method `Get`, you can define that interface here and add it to your `HttpApiClient` struct and work with it.